src/main/java/mate/academy/springboot/swagger/util/PaginationSortUtil.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.springboot.swagger.util;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PaginationSortUtil {
+    private static final int DIRECTION_INDEX = 1;
+    private static final int FIELD_INDEX = 0;
+    private static final String COLON = ":";
+    private static final String SEMICOLON = ";";
+
+    public PageRequest getPageRequest(Integer count, Integer page, String sortBy) {
+        List<Sort.Order> orders = new ArrayList<>();
+        if (sortBy.contains(SEMICOLON)) {
+            String[] sortingFields = sortBy.split(SEMICOLON);
+            for (String field: sortingFields) {
+                Sort.Order order;
+                if (field.contains(COLON)) {
+                    String[] fieldsAndDirections = field.split(COLON);
+                    order = new Sort
+                            .Order(Sort.Direction.valueOf(fieldsAndDirections[DIRECTION_INDEX]),
+                            fieldsAndDirections[FIELD_INDEX]);
+                } else {
+                    order = new Sort.Order(Sort.Direction.ASC, field);
+                }
+                orders.add(order);
+            }
+        } else {
+            Sort.Order order = new Sort.Order(Sort.Direction.ASC, sortBy);
+            orders.add(order);
+        }
```

Comment:
If your request contains only one sorting parameter and order (as an example: `sortBy=price:DESC`), it won't work, because you added this check: `if (sortBy.contains(SEMICOLON))` - so if the request doesn't contain `";"`, parsing into property and direction will not apply.